### DISCUSSION TITLE
"Sorry, no puzzle found" Error

### DISCUSSION THREAD

**Discussion by noahiwnl** [brianch/offline-chess-puzzles#14] [General]
Hey man, stumbled upon your Project and just a quick question because something does not add up. I followed the installing progress and put the "lichess_db_puzzle.csv.zst" file into the "puzzle" folder. However when booting up the application and trying to search after a puzzle, the message "Sorry, no puzzle found" pops up. Can you help me with a fix? Or am I just overlooking an step for the set up.

And if possible, how exactly can i add engines?

Thanks in advance!

**Comment by brianch** [comment]
Hello!

The .zst is like a .zip file, it's  a compressed archive, you need to extract it first (with 7-zip, peazip or any other file extractor, you most probably already have one installed).

About adding an engine, you just add the full path to the engine file, including the .exe name (something like: "C:\Users\myUser\Documents\stockfish_15.1.exe"), in the engine path field in the app settings, then you will be able to use the engine in the puzzle mode (switch using the selector below the board). In the future I can add a file searcher to make this step easier.

**Reply by noahiwnl** [reply]
So the egine works perfectly, however I am still stuck with the .zst file, I use 7zip but also just looked into Zstd(1.5.5). I was able to "convert" it into a .7z file but then my progress stopped. Any Tips? And thanks for the response!

**Reply by brianch** [reply]
Hmm, are you sure you didn't archive again the zst archive with 7-zip? What you want is to extract the zst, not compress.

From what I read, 7-zip has support for zst, but you could also try with another extractor, like Peazip.

The zstd project is more of a low-level thing (which these GUI apps would use to extract it e.g.), you don't really need to use it directly.

**Reply by noahiwnl** [reply]
Hmm, quite sure I did not. Another thing is that when I try to extract/open the file, 7zip tells me that it is unable to open the Archive.

**Reply by brianch** [reply]
Oh, maybe I got it wrong and it doesn't have support after all, try Peazip, that one I'm sure has support for zst.

**Reply by noahiwnl** [reply]
It worked! Thank you so much!

**Reply by brianch** [reply]
That's great to hear! Enjoy the puzzles :)
